USAGE.md

```diff
@@ -0,0 +1,365 @@
+
+func _on_block(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> void:
+    # Logic when the ability is blocked
+    # Typically interrupt the ability effect here
+    pass
+
+func _on_end(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> void:
+    # Logic when the ability ends
+    # Typically clean up the ability effect here
+    pass
+
+func _on_grant(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> void:
+    # Logic when the ability is granted
+    # Typically initialize the ability here
+    pass
+
+func _on_revoke(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> void:
+    # Logic when the ability is revoked
+    # Typically perform complete cleanup of the ability here
+    pass
+#endregion
+
+
+#region Condition check methods
+# These methods determine whether state transitions are allowed
+
+func _can_activate_cooldown(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> bool:
+    # Default: Always allow cooldown activation
+    return true
+
+func _can_be_activated(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> bool:
+    # Default: Always allow ability activation
+    return true
+
+func _can_be_blocked(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> bool:
+    # Default: Always allow ability blocking
+    return true
+
+func _can_be_granted(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> bool:
+    # Default: Always allow ability granting
+    return true
+
+func _can_be_revoked(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> bool:
+    # Default: Always allow ability revocation
+    return true
+
+func _can_be_ended(_ability_container: AbilityContainer, _runtime_ability: RuntimeAbility) -> bool:
+    # Default: Always attempt to end (if conditions permit)
+    return true
+#endregion
+
+#region Property retrieval methods
+# These methods provide dynamic properties of the ability
+
+func _get_cooldown(_ability_container: AbilityContainer) -> float:
+    # Default: No cooldown time
+    return 0.0
+
+func _get_duration(_ability_container: AbilityContainer) -> float:
+    # Default: No duration
+    return 0.0
+#endregion
+
+#region Automatic state management methods
+# These methods implement automated state management
+
+func _should_be_activated(_ability_container: AbilityContainer) -> bool:
+    # Default: Do not auto-activate
+    return false
+
+func _should_be_blocked(_ability_container: AbilityContainer) -> bool:
+    # Default: Do not auto-block
+    return false
+
+func _should_be_ended(_ability_container: AbilityContainer) -> bool:
+    # Default: Always try to end (if conditions allow)
+    return true
+#endregion
+
+#region Methods for cautious use
+# Overriding the _on_tick method will cause the cooldown and duration mechanisms to fail, as well as the per-frame _should_* method checks
```

Comment:
I would modify this as the same as for the .xml file, remembering that who overrides this function is in full control of the ticking of it.